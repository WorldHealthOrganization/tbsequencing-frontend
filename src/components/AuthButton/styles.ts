import { css } from '@emotion/react';
import { fontSizes } from '../typography/fontSizes';
import appColors from '../../styles/colors';
import { appZIndex } from '../../styles/appZIndex';

export const button = css({
  fontSize: fontSizes.h2,
  textTransform: 'capitalize',
  color: appColors.primary.navy,
});

export const nonClickableButton = css({
  pointerEvents: 'none',
  cursor: 'auto',
});

export const sxIcon = {
  fontSize: 40,
  fill: appColors.secondary.whoBlue,
};

export const mIcon = {
  fontSize: 32,
  fill: appColors.secondary.whoBlue,
};

export const iconWrapper = css({
  display: 'flex',
});

export const buttonWrapper = css({
  display: 'flex',
  position: 'relative',
});

export const active = css({
  borderBottom: `${4}px solid ${appColors.secondary.whoBlue}`,
});

export const fakeBorderBottom = css({
  height: 4,
  backgroundColor: appColors.secondary.whoBlue,
  position: 'absolute',
  left: 0,
  right: 0,
  bottom: 0,
  zIndex: appZIndex.appBar,
});
