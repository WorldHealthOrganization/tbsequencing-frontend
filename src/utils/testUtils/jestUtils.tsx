import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
