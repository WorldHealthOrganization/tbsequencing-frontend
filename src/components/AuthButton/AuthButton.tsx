/** @jsxImportSource @emotion/react */
import React from 'react';
import { Button } from '@mui/material';
import { AccountBox, Logout } from '@mui/icons-material';
import { SerializedStyles } from '@emotion/react';
import { useMsal } from '@azure/msal-react';
import { useIsLoggedIn } from '../../features/auth/hooks/useIsLoggedIn';
import * as styles from './styles';

interface Props {
  onClick?: () => void;
  label: string;
  isActive: boolean;
  shouldRenderBorderBottom?: boolean;
  style?: SerializedStyles;
  isAppBar?: boolean;
}

const FakeBorderBottom = () => <div css={styles.fakeBorderBottom} />;

export const AuthButton = ({
  label, onClick, isActive, shouldRenderBorderBottom = true, style, isAppBar,
}: Props) => {
  const { instance } = useMsal();
  const isLoggedIn = useIsLoggedIn();
  const activeAccount = instance.getAllAccounts();
  return (
    <div css={styles.buttonWrapper}>
      <Button
        css={[styles.button, style]}
        startIcon={<span css={styles.iconWrapper}><AccountBox sx={styles.sxIcon} /></span>}
        onClick={onClick}
        endIcon={(isAppBar && isLoggedIn && activeAccount[0])
        && <span css={styles.iconWrapper}><Logout sx={styles.mIcon} /></span>}
      >
        {label}
      </Button>
      {(isActive && shouldRenderBorderBottom) && <FakeBorderBottom />}
    </div>
  );
};

export default AuthButton;
