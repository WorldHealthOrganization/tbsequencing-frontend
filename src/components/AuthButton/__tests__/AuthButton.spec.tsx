import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../utils/testUtils/jestUtils';
import AuthButton from '../AuthButton';

describe('AuthButton component test', () => {
  const mockOnClick = jest.fn();
  const mockActive = true;
  const mockLabel = 'mock';

  test('renders AuthButton component without error', () => {
    render(
      <AuthButton label={mockLabel} onClick={mockOnClick} isActive={mockActive} />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <AuthButton label={mockLabel} onClick={mockOnClick} isActive={mockActive} />,
    );

    expect(tree).toMatchSnapshot();
  });
});
