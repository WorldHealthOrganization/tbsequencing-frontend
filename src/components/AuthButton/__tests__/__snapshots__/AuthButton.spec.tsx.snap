// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton component test Should render correctly 1`] = `
<div
  className="css-1y2xnzi-AuthButton"
>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1onmchy-MuiButtonBase-root-MuiButton-root-AuthButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <span
        className="css-9rttau-AuthButton"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1083c97-MuiSvgIcon-root"
          data-testid="AccountBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
          />
        </svg>
      </span>
    </span>
    mock
  </button>
  <div
    className="css-112tdqe-FakeBorderBottom"
  />
</div>
`;
